lib/bolt_server/transport_app.rb

```diff
@@ -727,6 +681,20 @@ def make_winrm_target(target_hash)
       [404, err.to_json]
     end
 
+    error BoltServer::RequestError do |err|
+      [400, err.to_json]
+    end
+
+    error Bolt::Error do |err|
+      # In order to match the request code pattern, unknown plan/task content should 400. This also
+      # gives us an opportunity to trim the message instructing users to use CLI to show available content.
+      if ['bolt/unknown-plan', 'bolt/unknown-task'].include?(err.kind)
+        [404, BoltServer::RequestError.new(err.msg.split('.').first).to_json]
```

Comment:
This feels a bit too clever. Could we just hardcode this message so we remove the dependency on `err`'s message? E.g. `Unknown task/plan ....`

Reply:
that would be more work because we dont actually have the requested plan/task (that data is only in the error message). One option would be to change this in bolt. Perhaps have the unknown task/plan error only give the message about the unknown requested content and then the CLI handler print CLI specific messaging about running `show`.

Reply:
👍 Using `err.msg` to extract the unknown task/plan since that is the only way to do so makes sense. I'd add a comment explaining that reasoning.

Re: options to fix, probably extending Bolt to include it in `details` should be enough for us b/c that's where I'd expect that info to be.